packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -284,13 +276,37 @@ export class SerializedStateManager {
 		}
 	}
 
+	private tryRefreshSnapshot(): void {
+		if (
+			this._refreshSnapshotP === undefined &&
+			this.mc.config.getBoolean("Fluid.Container.enableOfflineSnapshotRefresh") === true
+		) {
+			// Don't block on the refresh snapshot call - it is for the next time we serialize, not booting this incarnation
+			this._refreshSnapshotP = this.refreshLatestSnapshot(this.supportGetSnapshotApi());
+			this._refreshSnapshotP
+				.catch(
+					(error: TelemetryEventPropertyTypeExt | Tagged<TelemetryEventPropertyTypeExt>) => {
+						this.mc.logger.sendTelemetryEvent({
+							eventName: "RefreshLatestSnapshotFailed",
+							error,
+						});
+						this._refreshSnapshotP = undefined;
```

Comment:
doing this in the finally is enough. you don't need to do it in the catch as well